build/cmake/init.cmake

```diff
@@ -94,10 +94,10 @@ elseif(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") OR ("${CMAKE_CXX_COMPILER_ID}
     endif()
 endif()
 
-if(wxBUILD_COMPATIBILITY VERSION_LESS 3.2)
+if(wxBUILD_COMPATIBILITY STREQUAL "3.0")
```

Comment:
If users have `wxBUILD_COMPATIBILITY` `2.8` in their current configuration, this will now result in them having no compatibility at all.
I think it is better to move them to `3.0` by keeping the `VERSION_LESS` checks.
I'm not sure if you need a check for `NONE` or if you can rely on the version checks to fail.

Reply:
I have pushed yet another commit, with changes based on Maarten's comment.